src/odb/include/odb/db.h

```diff
@@ -7816,6 +7816,25 @@ class dbTechLayer : public dbObject
   ///
   int getSpacing(int width, int length = 0);
 
+  ///
+  /// Get the spacing between tracks for this layer.
+  /// If the layer has a multi pattern spacing returns the average.
+  ///
+  void getAverageTrackSpacing(odb::dbTrackGrid* track_grid,
+                              int& track_step,
+                              int& track_init,
+                              int& num_tracks);
+
+  ///
+  /// Populate parameters with average track information
+  /// for tracks with multiple patterns.
+  ///
+  void getAverageTrackPattern(odb::dbTrackGrid* grid,
```

Comment:
When would a user call this version?  It seems like a private internal method.

Reply:
Indeed, it makes sense to be private. The private methods are the ones starting with an underscore?

Reply:
Move the method to 'class _dbTechLayer' and remove it from db.h.  Be sure to put it in the "// User Code Begin Methods" block in dbTechLayer.h